lucene/core/src/java/org/apache/lucene/index/DocumentsWriterPerThreadPool.java

```diff
@@ -112,19 +110,12 @@ private synchronized DocumentsWriterPerThread newWriter() {
   DocumentsWriterPerThread getAndLock() {
     synchronized (this) {
       ensureOpen();
-      // Important that we are LIFO here! This way if number of concurrent indexing threads was once high,
-      // but has now reduced, we only use a limited number of DWPTs. This also guarantees that if we have suddenly
-      // a single thread indexing
-      final Iterator<DocumentsWriterPerThread> descendingIterator = freeList.descendingIterator();
-      while (descendingIterator.hasNext()) {
-        DocumentsWriterPerThread perThread = descendingIterator.next();
-        if (perThread.tryLock()) {
-          descendingIterator.remove();
-          return perThread;
-        }
+      DocumentsWriterPerThread dwpt = freeList.poll(DocumentsWriterPerThread::tryLock);
+      if (dwpt == null) {
+        // DWPT is already locked before return by this method:
```

Comment:
Hmm, can you put this comment back in its original place (just before the `return dwpt`?  It is confusing here, making me think the "allocate a new DWPT" case has something to do with the locking semantics.

Reply:
> making me think the "allocate a new DWPT" case has something to do with the locking semantics.

Hmm, this is exactly what my understanding is. :) To me the comment wat about highlighting that `newWriter()` implicitly takes the lock on the DWPT it creates?